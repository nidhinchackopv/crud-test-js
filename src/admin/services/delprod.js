export async function delData(id) {
    const data = await fetch(`http://localhost:3000/users/${id}`,{
                 method: 'DELETE'
            });
        return data;
}