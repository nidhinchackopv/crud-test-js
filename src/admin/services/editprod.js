

export async function editProd(props,state) {
    console.log(props)
    console.log(state)
    const data = await fetch(`http://localhost:3000/users/${props.match.params.id}`, {
            method: 'PUT',
            headers: {'Content-Type':'application/json'},
            body: JSON.stringify({
                title: state.title,
                body: state.body
            })
        })
 return data;
}

export async function editProdOne(props) {
    console.log(props)
        const response = await fetch(`http://localhost:3000/users/${props.match.params.id}`);

        const data = await response.json();
        return data;
}