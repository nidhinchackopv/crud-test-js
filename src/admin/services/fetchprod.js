export async function fetchData() {

    const response = await fetch("http://localhost:3000/users")
    const data = await response.json();
    return data;
}