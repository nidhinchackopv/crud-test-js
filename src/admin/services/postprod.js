export async function getData(title,body) {
    const data = await fetch("http://localhost:3000/users", {
            method: 'POST',
            headers: {'Content-Type':'application/json'},
            body: JSON.stringify({
                title,
                body
            })
        })
 return data;
}