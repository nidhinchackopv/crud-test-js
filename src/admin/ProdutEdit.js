import React, { PropsWithoutRef, SyntheticEvent, useState, useEffect } from 'react'
import { Redirect } from 'react-router-dom';
import * as APIEdit from './services/editprod'

const ProductEdit = (props) => {
    const [state, setState] = useState([{title:"",body:""}]);
    const [redirect, setRedirect] = useState(false);

console.log(props)
    const hello = async (props) => {
        console.log(props)
        const data = await APIEdit.editProdOne(props);

        setState({title:data.title,body:data.body});
    }
    useEffect( () =>{
       hello(props);
    },[]);

    const submit = async (e) => {
        console.log(e)
        console.log(props)
        e.preventDefault();

       await APIEdit.editProd(props,state)

        setRedirect(true);
        
    }

    if(redirect){
        return(
            <Redirect to={'/home'}/>
        )
    }
console.log(state)
    return (
        
        <form data-testid="pro-form" onSubmit={(e)=>submit(e)} className="px-4 rounded mx-auto pt-10 pb-10 shadow-md border-solid rounded-xl max-w-3xl w-full my-32 inputs space-y-6 bg-gray-200">
        <div className="flex space-x-4">
                <div className="w-1/2 flex ">
                    <label className="pr-2 pt-2 text-center">Title</label>
                    <input data-testid="pro-input" type="text" className="ml-12 border border-gray-400 px-4 py-2 rounded w-full focus:outline-none focus:border-teal-400" name="Title"
                    defaultValue={state.title}
                    onChange={(e) => setState({...state,title:e.target.value})}/>
                </div>
                </div>
                <div className="flex">
                    <label className="pr-2">Description</label>
                    <textarea data-testid="pro-textarea" className="border border-gray-400 px-4 py-2 rounded w-full focus:outline-none focus:border-teal-400" name="body"
                    defaultValue={state.body}
                    onChange={(e) =>  setState({...state,body:e.target.value})}/>
                </div>
                <button data-testid="submit" type="submit" className="shadow-md border-solid rounded-xl border-4 border-gray-200 bg-purple-500 p-2">Submit</button>
            </form>

    )
}

export default ProductEdit
