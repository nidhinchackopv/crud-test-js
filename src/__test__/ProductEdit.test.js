import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import * as ReactDom from 'react-dom'
import ProdutEdit from '../admin/ProdutEdit'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import * as APIEdit from '../admin/services/editprod';


// let state={title:"hello",body:"world"}
jest.mock('../admin/services/editprod');
it('fetching', async () => {
  APIEdit.editProd.mockResolvedValueOnce({ok : true});
  render(<ProdutEdit />);
  const input = screen.getByTestId('pro-input');
  fireEvent.change(input, { 'target' : { 'value' : 'hello' }});
  const textarea = screen.getByTestId('pro-textarea');
  fireEvent.change(textarea, { 'target' : { 'value' : 'world' }});
  const button = screen.getByTestId('submit');
  fireEvent.click(button);
  expect(APIEdit.editProd).toHaveBeenCalledTimes(1)
  // expect(APIEdit.editProd).toHaveBeenCalledWith(1 , state)
})

jest.mock('../admin/services/editprod');

it('fetching', async () => { 
  APIEdit.editProdOne.mockResolvedValueOnce({ok : true});
  render(<ProdutEdit />);
  expect(APIEdit.editProdOne).toHaveBeenCalledTimes(1)
})


it('can render the About page', () => {
  const tree = renderer.create(<ProdutEdit />).toJSON();
  expect(tree).toMatchSnapshot();
})

describe("component test", () => {
  let container

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    ReactDom.render(<ProdutEdit />, container);
  })

  afterEach(() => {
    document.body.removeChild(container);
    container.remove();
  })

  it("contains the ellements", () => {
    expect(container.querySelector("[data-testid = 'pro-input']")?.getAttribute('name')).toBe('Title');
    expect(container.querySelector("[data-testid = 'pro-textarea']")?.getAttribute('name')).toBe('body');
  })

  test("table items", () => {
    const button = screen.getByTestId("submit");
    fireEvent.click(button);
    expect(screen.getByText("Title"))
    expect(screen.getByText("Description"))
  })

  test("should display a form with values ", () => {
    const input = screen.getByTestId("pro-input");
    fireEvent.change(input);
    const textarea = screen.getByTestId("pro-textarea");
    fireEvent.change(textarea);
    expect(screen.getByTestId("pro-form").getAttribute('Title','body'))
  });

  // test("should display a form with values ", () => {
  //   const defaultValue = "defaultValue";
  //   ReactDom.render(<ProdutEdit initial={defaultValue}/>, container);
  //   expect(container.getByTestId(/pro-input/i).value).toBe(defaultValue);
  // });


})
