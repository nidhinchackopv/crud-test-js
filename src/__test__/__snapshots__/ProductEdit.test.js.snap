// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render the About page 1`] = `
<form
  className="px-4 rounded mx-auto pt-10 pb-10 shadow-md border-solid rounded-xl max-w-3xl w-full my-32 inputs space-y-6 bg-gray-200"
  data-testid="pro-form"
  onSubmit={[Function]}
>
  <div
    className="flex space-x-4"
  >
    <div
      className="w-1/2 flex "
    >
      <label
        className="pr-2 pt-2 text-center"
      >
        Title
      </label>
      <input
        className="ml-12 border border-gray-400 px-4 py-2 rounded w-full focus:outline-none focus:border-teal-400"
        data-testid="pro-input"
        name="Title"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="flex"
  >
    <label
      className="pr-2"
    >
      Description
    </label>
    <textarea
      className="border border-gray-400 px-4 py-2 rounded w-full focus:outline-none focus:border-teal-400"
      data-testid="pro-textarea"
      name="body"
      onChange={[Function]}
    />
  </div>
  <button
    className="shadow-md border-solid rounded-xl border-4 border-gray-200 bg-purple-500 p-2"
    data-testid="submit"
    type="submit"
  >
    Submit
  </button>
</form>
`;
